cryptarchia/cryptarchia.py

```diff
@@ -420,10 +417,10 @@ def try_create_fork(self, block: BlockHeader) -> Optional[Chain]:
 
         chains = self.forks + [self.local_chain]
         for chain in chains:
-            if chain.contains_block(block):
-                block_position = chain.block_position(block)
+            block_position = chain.block_position(block.parent)
+            if block_position != -1:
                 return Chain(
-                    blocks=chain.blocks[:block_position],
+                    blocks=chain.blocks[: block_position + 1],
```

Comment:
@davidrusu @zeegomo I just added a test that covers all cases of fork creation. 2460827

Previously, this case hasn't been caught by tests because we've only tested the case which a fork is created from the genesis block, not from the middle of the chain. https://github.com/logos-co/nomos-specs/blob/b1e13f79c5e4ddefbd8424a3188bc093c355d234/cryptarchia/test_ledger_state_update.py#L44